code/modules/mob/language.dm

```diff
@@ -41,9 +43,42 @@
 
 	return "[trim(full_name)]"
 
-/datum/language/proc/scramble(input)
+/datum/language/proc/scramble(input, list/known_languages)
 
-	if(!syllables || !syllables.len)
+	var/understand_chance = 0
+	for(var/datum/language/L in known_languages)
+		if(LAZYACCESS(partial_understanding, L.type))
+			understand_chance += partial_understanding[L.type]
+		if(LAZYACCESS(L.partial_understanding, src.type))
+			understand_chance += L.partial_understanding[type] * 0.5
+
+	var/scrambled_text = ""
+	var/list/words = splittext(input, " ")
+	for(var/w in words)
+		if(prob(understand_chance))
+			scrambled_text += " [w] "
+		else
+			var/nword = scramble_word(w)
+			var/ending = copytext(scrambled_text, length(scrambled_text) - 1)
+			if(findtext(ending, "."))
+				nword = capitalize(nword)
+			scrambled_text += nword
+	scrambled_text = replacetext(scrambled_text, "  ", " ")
+
+	scrambled_text = capitalize(scrambled_text)
+	scrambled_text = trim(scrambled_text)
+	var/ending = copytext(scrambled_text, length(scrambled_text))
+	if(ending == ".")
+		scrambled_text = copytext(scrambled_text, 1, length(scrambled_text) - 1)
+
+	var/input_ending = copytext(input, length(input))
+	if(input_ending in list("!", "?", "."))
+		scrambled_text += input_ending
+
+	return scrambled_text
+
+/datum/language/proc/scramble_word(var/input)
```

Comment:
Linters is failing here. This argument cannot start with `var`. I think you will be okay with just using `input`.